Change - Add apache::mod_status class

blackknight36 opened the pull request:
This class enables and configures the apache status module.q

Commit:
Change - Add apache::mod_status class

This class enables and configures the apache status module.

jflorian commented on README.md:
```diff
@@ -116,6 +117,19 @@ An array of package names needed for httpd Phusion Passenger support.  The defau
+only be allowed on the local loopback interface.
+
+##### `location`
+This parameter controls the location used to access the apache status page.  Default value is set to /server-status.
```
Can you please change this so that default is represented as code, specifically a quoted string.  That way it'll be consistent, see one of the *ensure* parameters.

jflorian commented on README.md:
```diff
@@ -116,6 +117,19 @@ An array of package names needed for httpd Phusion Passenger support.  The defau
+This parameter controls the location used to access the apache status page.  Default value is set to /server-status.
+
+##### `allow_from`
+An array of IPs or IP ranges in CIDR format which define what hosts are allowed to access the server status page.
```
Does this have a default?  If not, please annotate as `(required)` like I've done elsewhere, again for consistency.

blackknight36 commented:
I've updated the readme file as requested.

jflorian commented:
A few more things:

1. The `manage_firewall` param doc in the README is now under `mod_status` rather than `mod_ssl` to which it belongs.

1. Also, I tried testing a deployment and got: `Error: /Stage[main]/Apache::Mod_status/File[/etc/httpd/conf.d/mod_status.conf]: Could not evaluate: Could not retrieve information from environment production source(s) puppet:///modules/apache/mod_status.conf`.  It looks like this is because the manifest has a source URI, but the module is providing and needs a template.  

1. Please rename the template to *.erb.

blackknight36 commented:
Thanks for catching that.  I fixed the template file and moved the
parameter documentation to the proper sections.  The class should apply
cleanly now as well.


On 5/21/19 1:50 PM, John Florian wrote:
>
> A few more things:
>
> 1.
>
>     The |manage_firewall| param doc in the README is now under
>     |mod_status| rather than |mod_ssl| to which it belongs.
>
> 2.
>
>     Also, I tried testing a deployment and got: |Error:
>     /Stage[main]/Apache::Mod_status/File[/etc/httpd/conf.d/mod_status.conf]:
>     Could not evaluate: Could not retrieve information from
>     environment production source(s)
>     puppet:///modules/apache/mod_status.conf|. It looks like this is
>     because the manifest has a source URI, but the module is providing
>     and needs a template.
>
> 3.
>
>     Please rename the template to *.erb.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/jflorian/doubledog-apache/pull/1?email_source=notifications&email_token=AEJPK5Z5STPOU7XQ673X7XDPWQY7FA5CNFSM4HKGR5S2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODV4VS2Y#issuecomment-494492011>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEJPK52G532PP457K2KSGCDPWQY7FANCNFSM4HKGR5SQ>.
>
